packages/ui/src/context/RouterContext.tsx

```diff
@@ -0,0 +1,67 @@
+import { ComponentType, createContext, ReactNode, useContext } from 'react'
+import type { LinkProps, NavLinkProps, OutletProps } from 'react-router-dom'
+
+import { cn } from '@utils/cn'
+
+interface RouterContextType {
+  Link: ComponentType<LinkProps>
+  NavLink: ComponentType<NavLinkProps>
+  Outlet: ComponentType<OutletProps>
+  navigate: (to: string, options?: { replace?: boolean }) => void
+}
+
+const resolveTo = (to: LinkProps['to']) => (typeof to === 'string' ? to : to.pathname || '/')
+
+const LinkDefault = ({ to, children, className, ...props }: LinkProps) => {
+  const href = resolveTo(to)
+  return (
+    <a href={href} className={cn('text-blue-500 hover:underline', className)} {...props}>
+      {children}
+    </a>
+  )
+}
+
+const NavLinkDefault = ({ to, children, className, style, ...props }: NavLinkProps) => {
+  const href = resolveTo(to)
+  const isActive = new URL(href, window.location.origin).pathname === window.location.pathname
+
+  const finalClassName =
+    typeof className === 'function'
+      ? className({ isActive, isPending: false, isTransitioning: false })
+      : cn(className, isActive ? 'font-bold text-blue-600' : 'text-blue-500')
```

Comment:
do we really need to include custom styles in the fallback default impl?

Reply:
Not needed, will clean it up.